src/map/status.c

```diff
@@ -5051,19 +5051,20 @@ static unsigned short status_calc_speed(struct block_list *bl, struct status_cha
 	if( sc == NULL || ( sd && sd->state.permanent_speed ) )
 		return (unsigned short)cap_value(speed,MIN_WALK_SPEED,MAX_WALK_SPEED);
 
-	if (sd && sd->ud.skilltimer != INVALID_TIMER)
-	{
-		if (sd->ud.skill_id == LG_EXEEDBREAK) {
-			speed_rate = 160 - 10 * sd->ud.skill_lv;
-		} else if ((skill->get_inf2(sd->ud.skill_id) & INF2_FREE_CAST_REDUCED) != 0) {
-			speed_rate = 175 - 5 * sd->ud.skill_lv;
-		} else if (pc->checkskill(sd, SA_FREECAST) > 0) {
-			speed_rate = 175 - 5 * pc->checkskill(sd,SA_FREECAST);
-		}
-	}
+	bool skillTimerIsValid = sd && sd->ud.skilltimer != INVALID_TIMER;
```

Comment:
in all lines you forgot to add check for ``sd`` non null (``sd != NULL``). Only if sure what ``sd`` is not null it can be used as pointer

Reply:
are you saying there is a bug in the existing logic ?  Because I didn't change that.  `sd && sd->ud.skilltimer != INVALID_TIMER` would resolve to false if sd is null, correct ?

Reply:
here's a fiddle showing sd as NULL should work fine

https://www.mycompiler.io/view/Hzv9nGJ

Reply:
ah i not saw this one check for null. Then it can works
but still better check it like ``sd != NULL && sd-> ...`` for follow style
and some missing spaces below.

Reply:
I can add the null check but I don't see any mention of it in the style guide.

Also what spaces am I missing ?  I'm not seeing anything missing [per the whitespace section](https://github.com/HerculesWS/Hercules/wiki/Coding-Style#32-spaces)

Reply:
``sd != NULL`` because ``sd`` is not bool and better check it with some non bool value. and for pointers 0 is NULL.

for missing spaces this ``pc->checkskill(sd,SA_FREECAST)`` -> ``pc->checkskill(sd, SA_FREECAST)``

Reply:
Thanks for explaining the null style, I'll try to keep types consistent.  And thanks for pointing out the space